db/db.go

```diff
@@ -2892,6 +2926,53 @@ func GetTotalBLSChanges() (uint64, error) {
 	return count, nil
 }
 
+func GetBLSChangesCountForQuery(query string) (uint64, error) {
+	count := uint64(0)
+
+	blsQuery := `
+		SELECT COUNT(*)
+		FROM blocks_bls_change bls
+		INNER JOIN blocks b ON bls.block_root = b.blockroot AND b.status = '1'
+		%s
+		%s`
+
+	trimmedQuery := strings.ToLower(strings.TrimPrefix(query, "0x"))
+	var err error = nil
+
+	joinQuery := `
+		LEFT JOIN (
+			SELECT 
+				validators.validatorindex as validatorindex,
+				eth1_deposits.from_address as deposit_adress
+			FROM validators 
+			INNER JOIN eth1_deposits ON validators.pubkey = eth1_deposits.publickey
+		) AS val ON val.validatorindex = bls.validatorindex`
+
+	// Check whether the query can be used for a validator, slot or epoch search
+	if uiQuery, parseErr := strconv.ParseUint(query, 10, 64); parseErr == nil {
+		searchQuery := `
+			WHERE bls.validatorindex = $1			
+				OR block_slot = $1
+				OR (block_slot / $3) = $1
+				OR pubkey_text LIKE ($2 || '%')
+				OR ENCODE(val.deposit_adress, 'hex') LIKE ($2 || '%')`
```

Comment:
That's not how we spell address in English, even if you just copied it from GetBLSChanges, I would correct it.

Reply:
renamed all occurrences of deposit_adress